### PR TITLE
Sous 824 structured logging on singularity

### PR DESCRIPTION
**Pull request by lachlanh**

### CODE REVIEW

File: ext/singularity/actual_deployment_set.go [hunk 1]
```diff
@@ -186,21 +185,25 @@ func (sc *deployer) singPipeline(
 	errs chan error,
 	clusters sous.Clusters,
 ) {
-	sc.log.Vomitf("Starting cluster at %s", url)
-	defer func() { sc.log.Vomitf("Completed cluster at %s", url) }()
+	messages.ReportLogFieldsMessage("Starting Cluster", logging.DebugLevel, sc.log, url)
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: before this was a ExtraDebug1

File: ext/singularity/actual_deployment_set.go [hunk 2]
```diff
@@ -186,21 +185,25 @@ func (sc *deployer) singPipeline(
...
+	defer func() { messages.ReportLogFieldsMessage("Completed Cluster", logging.DebugLevel, sc.log, url) }()
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: Same here

File: ext/singularity/actual_deployment_set.go [hunk 3]
```diff
@@ -186,21 +185,25 @@ func (sc *deployer) singPipeline(
...
 	defer wg.Done()
 	defer catchAndSend(fmt.Sprintf("get requests: %s", url), errs, sc.log)
 	srp, err := sc.getSingularityRequestParents(client)
 	if err != nil {
-		sc.log.Vomitf("%v", err) //XXX connection reset by peer should be retried
+		messages.ReportLogFieldsMessage("Error in singPipeline", logging.DebugLevel, sc.log, err)
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: Same

File: ext/singularity/actual_deployment_set.go [hunk 4]
```diff
@@ -186,21 +185,25 @@ func (sc *deployer) singPipeline(
...
 	for _, r := range rs {
-		sc.log.Vomitf("Req: %s %s %d", r.SourceURL, reqID(r.ReqParent), r.ReqParent.Request.Instances)
+		messages.ReportLogFieldsMessage("Request",
+			logging.DebugLevel,
+			sc.log, r.SourceURL,
+			reqID(r.ReqParent),
+			r.ReqParent.Request.Instances)
 		dw.Add(1)
 		reqs <- r
 	}
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: same

File: ext/singularity/actual_deployment_set.go [hunk 5]
```diff
@@ -234,13 +237,13 @@ func (sc *deployer) depPipeline(
 	poolLimit := make(chan struct{}, poolCount)
 	for req := range reqCh {
 		depAssWait.Add(1)
-		sc.log.Vomitf("starting assembling for %q", reqID(req.ReqParent))
+		messages.ReportLogFieldsMessage("started assembling", logging.DebugLevel, sc.log, reqID(req.ReqParent))
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: same

File: ext/singularity/actual_deployment_set.go [hunk 6]
```diff
@@ -234,13 +237,13 @@ func (sc *deployer) depPipeline(
...
 		go func(req SingReq) {
 			defer depAssWait.Done()
 			defer catchAndSend(fmt.Sprintf("dep from req %s", req.SourceURL), errCh, sc.log)
 			poolLimit <- struct{}{}
 			defer func() {
-				sc.log.Vomitf("finished assembling for %q", reqID(req.ReqParent))
+				messages.ReportLogFieldsMessage("finished assembling", logging.DebugLevel, sc.log, reqID(req.ReqParent))
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: same

File: ext/singularity/actual_deployment_set.go (hunk 5 above)
Comment (JC1738) [reply ext/singularity/actual_deployment_set.go]: If this is an ID, can also switch these to ReportLogFieldsMessageWithIDs

File: ext/singularity/actual_deployment_set.go [hunk 7]
```diff
@@ -256,8 +259,8 @@ func (sc *deployer) depPipeline(
 }
 
 func (sc *deployer) assembleDeployState(reg sous.Registry, clusters sous.Clusters, req SingReq) (*sous.DeployState, error) {
-	sc.log.Vomitf("Assembling from: %s %s", req.SourceURL, reqID(req.ReqParent))
+	messages.ReportLogFieldsMessage("Assembling deploy state", logging.DebugLevel, sc.log, req.SourceURL, reqID(req.ReqParent))
```
Comment (JC1738) [inline ext/singularity/actual_deployment_set.go]: check log level, also could use ID logger

File: ext/singularity/deployment_builder.go [hunk 8]
```diff
@@ -56,10 +57,10 @@ func (nsd nonSousError) Error() string {
 }
 
 func ignorableDeploy(log logging.LogSink, err error) bool {
-	log.Vomitf("checking to see if %+v is ignorable", err)
+	messages.ReportLogFieldsMessage("checking to see if error is ignorable", logging.DebugLevel, log, err)
```
Comment (JC1738) [inline ext/singularity/deployment_builder.go]: ExtraDebug lvl

File: ext/singularity/deployment_builder.go [hunk 9]
```diff
@@ -56,10 +57,10 @@ func (nsd nonSousError) Error() string {
...
 	switch errors.Cause(err).(type) {
 	case nonSousError, notThisClusterError:
-		log.Vomitf("ignorable: %v", err)
+		messages.ReportLogFieldsMessage("error is ignorable", logging.DebugLevel, log, err)
```
Comment (JC1738) [inline ext/singularity/deployment_builder.go]: ExtraDebug lvl

**Comment by sous-machine-account** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by lachlanh** [comment]
@JC1738 have made the changes, and a small fix to check for nil on a type of, could you take another quick look and merge if everything is ok ?

**Pull request merged**